matchms/similarity/BaseSimilarity.py

```diff
@@ -231,13 +296,19 @@ def _sparse_array_with_mask_without_filter(self, references: Sequence[SpectrumTy
             scores[i] = self.pair(references[i_row], queries[i_col])
         return COOMatrix(row_idx=mask_indices.idx_row, column_idx=mask_indices.idx_col, scores=scores, scores_dtype=self.score_datatype)
 
-    def _sparse_array_with_filter(self, references: Sequence[SpectrumType], queries: Sequence[SpectrumType],
-                                 mask_indices: COOIndex) -> COOMatrix:
-        """Uses a mask to compute only the required scores and filters scores that do not pass the filter.
+    def _sparse_array_with_filter(
+            self,
+            references: Sequence[SpectrumType],
+            queries: Sequence[SpectrumType],
+            mask_indices: COOIndex) -> COOMatrix:
+        """
+        Compute a sparse similarity matrix (COO format) using a mask and applying score filters.
 
-        This method most of the time does not make sense. It is only worth it if you want to store less than 1/12th
-        of the computed scores. This is not the case most of the time, so sparse_array is most of the time the most
-        memory efficient.
+        Only the (row, column) pairs specified in mask_indices are computed, and only scores that pass
+        all filters are stored.
+
+        Please note: If the mask_indices contain more than 1/12th of all indices, this method could become less memory efficient
```

Comment:
This is slightly different. If the score filter removes less than 1/12th of all the pairs (in the mask_indiced), it is less memory efficient than just computing and storing all pairs without filtering and applying the filter after. 

It might make more sense to actually re-implement this method, to just run _sparse_array_with_mask_without_filter followed by filtering out based on the score filters. 
This saves quite some memory, since no lists are needed and the mask_indices are just reused instead of added again to a list.